src/transformers/tokenization_bart.py

```diff
@@ -0,0 +1,36 @@
+# coding=utf-8
+# Copyright 2020 The Facebook AI Research Team Authors and The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .tokenization_roberta import RobertaTokenizer
+
+
+# vocab and merges same as roberta
+vocab_url = "https://s3.amazonaws.com/models.huggingface.co/bert/roberta-large-vocab.json"
+merges_url = "https://s3.amazonaws.com/models.huggingface.co/bert/roberta-large-merges.txt"
+_all_bart_models = [
+    "bart-large",
+    "bart-large-mnli",
+    # "bart-large-cnn"
+]
+
+
+class BartTokenizer(RobertaTokenizer):
```

Comment:
This `BartTokenizer` seems to get different results comparing to the fairseq one: 

See https://github.com/pytorch/fairseq/blob/master/fairseq/models/bart/hub_interface.py#L59-L60

```
# fairseq
>>> bart.encode('Hello world').tolist()
[0, 31414, 232, 2]
>>> bart.encode(' world').tolist()
[0, 232, 2]
>>> bart.encode('world').tolist()
[0, 8331, 2]

# huggingface
tokenizer = BartTokenizer.from_pretrained('bart-large')
>>> tokenizer.encode("Hello world")
[0, 20920, 232, 2]
>>> tokenizer.encode(" world")
[0, 232, 2]
>>> tokenizer.encode("world")
[0, 232, 2]
```

It seems to be stripping spaces.

Reply:
Great catch! Will fix and add a test!

Reply:
Shot in the dark @joeddav. Is this related to the bug you found?

Reply:
larger issue than this PR. Filing it, but won't fix here.